libcalico-go/lib/backend/etcdv3/etcdv3.go

```diff
@@ -568,3 +611,71 @@ func parseRevision(revs string) (int64, error) {
 	}
 	return rev, nil
 }
+
+func storePolicyName(name string, annotations map[string]string) (map[string]string, error) {
+	metadata := map[string]string{}
+	metadata["name"] = name
+
+	metadataBytes, err := json.Marshal(metadata)
+	if err != nil {
+		return nil, err
+	}
+
+	if annotations == nil {
+		annotations = map[string]string{}
+	}
+	annotations[metadataAnnotation] = string(metadataBytes)
+
+	return annotations, nil
+}
+
+func defaultPolicyName(d *model.KVPair) error {
+	if _, ok := d.Value.(*apiv3.NetworkPolicy); ok {
+		value := d.Value.(*apiv3.NetworkPolicy)
+
+		annotations, err := storePolicyName(value.Name, value.Annotations)
+		if err != nil {
+			return err
+		}
+
+		value.Annotations = annotations
```

Comment:
```suggestion
                // First, capture the policy name that was used on the v3 API and store it as an annotation. 
                // This allows us to recreate the original object in List() and Watch() calls correctly, returning the object as expected by the client. 
		annotations, err := storePolicyName(value.Name, value.Annotations)
		if err != nil {
			return err
		}
		value.Annotations = annotations
```

Here's an example of the type of comment I'm looking for.